idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/intention/IntentionSyntax.kt

```diff
@@ -14,54 +15,107 @@ import org.jetbrains.kotlin.idea.quickfix.KotlinSingleIntentionActionFactory
 import org.jetbrains.kotlin.idea.quickfix.QuickFixContributor
 import org.jetbrains.kotlin.psi.KtElement
 
-interface IntentionExtensionProviderSyntax : IntentionExtensionProviderUtilitySyntax {
+/**
+ * The IDE analysis user code and provides [IntentionAction]'s to either signal error's to user's or resolve them if triggered.
+ */
+interface IntentionSyntax : IntentionUtilitySyntax {
+
+  /**
+   * registers Intention with Metadata such as Html description of the specific behavior of this [intention]
+   * @param category are used to order Intentions
+   * TODO: This Bails if there is no html and intentionDescription. Try to add them in the Function and not with the resource Folder. But this is fine for now
+   */
   fun IdeMetaPlugin.addIntentionWithMetaData(
+    category: String,
     intention: IntentionAction
   ): ExtensionPhase =
-    IntentionExtensionProvider.RegisterIntentionWithMetaData(intention)
+    IntentionExtensionProvider.RegisterIntentionWithMetaData(intention, category)
 
   /**
-   * TODO: This Bails if there is no html and intentionDescription. Try to add them in the Function and not with the resource Folder. But this is fine for now
+   * registers [intention]
    */
   fun IdeMetaPlugin.addIntention(
-    category: String,
     intention: IntentionAction
   ): ExtensionPhase =
-    IntentionExtensionProvider.RegisterIntention(intention, category)
+    IntentionExtensionProvider.RegisterIntention(intention)
 
+  /**
+   * unregisters [intention] from the editor
+   */
   fun IdeMetaPlugin.unregisterIntention(
     intention: IntentionAction
   ): ExtensionPhase =
     IntentionExtensionProvider.UnregisterIntention(intention)
 
+  /**
+   * registers an Intention and allows to manipulate the editor environment or [K]
```

Comment:
What does `[K]` stand for?

Reply:
[K] is the TypeParameter for this function, in this case any KtElement.